Sherlock-Holo/fuse3
# Fix dependency version specs

Also, make the following CI changes:

* Test with -Zdirect-minimal-versions
* Use the cargo cache
* Use "cargo check" instead of "cargo build", since there are no tests anyway
